test/celeritas/ext/SimpleSensitiveDetector.cc

```diff
@@ -91,14 +91,18 @@ bool SimpleSensitiveDetector::ProcessHits(G4Step* step, G4TouchableHistory*)
     auto* pre_step = step->GetPreStepPoint();
     CELER_ASSERT(pre_step);
 
-    hits_.energy_deposition.push_back(step->GetTotalEnergyDeposit()
-                                      / CLHEP::MeV);
     hits_.step_length.push_back(step->GetStepLength() / CLHEP::cm);
+
+    real_type edep = step->GetTotalEnergyDeposit();
```

Comment:
Convert to the native unit system here, rather than when appending to the vector. Generally, anything inside celeritas should have a native type or a Quantity to specify its unit system. Also, if we're running with `real_type = float`, then the implementation here is `float( float(edep) /  double(MeV))` rather than `float(double(edep) / deouble(MeV))` so there's some precision loss.